### ISSUE TITLE
Super Productivity won't sync properly with Dropbox when the date has changed on one device

### ISSUE THREAD

**Issue by Tim676o** [johannesjo/super-productivity#3160]
### Steps to Reproduce

1. Change the date to a future one on an iOS phone.
2. Create some simple counters on the website, on the phone.
3. Reset to the original date
4. When synced with dropbox, you can see on Windows Super Productivity app and iOS web app an error : "Sync Error: one of the dates provided is from the future..." and then a prompt asking if you want to import the distant data or not.
5. If you accet to import the data, it takes you back to a setup of super productivity before the change of date on the iOS phone, wether on Windows or iOS, and it continues forever to display the same message if you try to sync, even if you imported the data. It doesn't change anything to restart the app.

### Console Output

![Capture d'écran 2024-04-20 144558](https://github.com/johannesjo/super-productivity/assets/110531482/19e2ac0d-1a10-45d3-ad46-716e52ee71d9)

### Meta Info
META: SP8.0.5 Electron – fr – Win32 – Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) superProductivity/8.0.5 Chrome/122.0.6261.156 Electron/29.2.0 Safari/537.36

**Comment by Jagdfalke** [comment]
I'm not sure if I fully understand the issue here. In what scenario would a date change on a device?

**Comment by Tim676o** [comment]
> I'm not sure if I fully understand the issue here. In what scenario would a date change on a device?

I don't remember why I changed the date, maybe to check if something was available on some website if I brought forward the date. But now the bug seems to occur even when I did nothing. It occurs apparently from times to times. Simply relaunching the app solves it. I was afraid to do it because of potential data losses, but it wasn't harmful and sync is now working again.
